src/xcode/ENA/ENA/Source/Scenes/App/AppOnboardingNavigationController.swift

```diff
@@ -0,0 +1,153 @@
+
+		if let opacityDelegate = viewController as? NavigationBarOpacityDelegate {
+			navigationBackgroundAlpha = opacityDelegate.backgroundAlpha
+			largeTitleBlurEffect = opacityDelegate.preferredLargeTitleBlurEffect
+			largeTitleBackgroundColor = opacityDelegate.preferredLargeTitleBackgroundColor
+
+			observeScrollView(of: viewController)
+		}
+
+		let previousNavigationBackgroundAlpha = navigationBar.backgroundAlpha
+		if #available(iOS 13.0, *) {
+			let previousScrollEdgeAppearance = navigationBar.scrollEdgeAppearance
+			
+			transitionCoordinator?.animate(alongsideTransition: { _ in
+				self.navigationBar.backgroundAlpha = navigationBackgroundAlpha
+
+				if let largeTitleBackgroundColor = largeTitleBackgroundColor {
+					self.navigationBar.scrollEdgeAppearance = UINavigationBarAppearance()
+					self.navigationBar.scrollEdgeAppearance?.backgroundColor = largeTitleBackgroundColor
+				} else if let largeTitleBlurEffect = largeTitleBlurEffect {
+					self.navigationBar.scrollEdgeAppearance = UINavigationBarAppearance()
+					self.navigationBar.scrollEdgeAppearance?.backgroundEffect = UIBlurEffect(style: largeTitleBlurEffect)
+				} else {
+					self.navigationBar.scrollEdgeAppearance = self.defaultScrollEdgeAppearance
+				}
+
+			}, completion: { context in
+				if context.isCancelled {
+					self.navigationBar.backgroundAlpha = previousNavigationBackgroundAlpha
+					self.navigationBar.scrollEdgeAppearance = previousScrollEdgeAppearance
+
+					self.scrollViewObserver?.invalidate()
+					self.scrollViewObserver = previousScrollViewObserver
+
+				} else {
+					previousScrollViewObserver?.invalidate()
+				}
+			})
+		} else {
+			transitionCoordinator?.animate(alongsideTransition: { _ in
+				self.navigationBar.backgroundAlpha = navigationBackgroundAlpha
+
+			}, completion: { context in
+				if context.isCancelled {
+					self.navigationBar.backgroundAlpha = previousNavigationBackgroundAlpha
+
+					self.scrollViewObserver?.invalidate()
+					self.scrollViewObserver = previousScrollViewObserver
+
+				} else {
+					previousScrollViewObserver?.invalidate()
+				}
+			})
+		}
+
+	}
+
+	/// If the passed in `UIViewController` is a `NavigationBarOpacityDelegate` and contains a `UIScrollView`,
+	/// register an observer for the `contentOffset` property so that the navigation bar's `backgroundAlpha` is set as the controller scrolls.
+	///
+	/// - parameter viewController: The controller to regsiter the observer for
+	private func observeScrollView(of viewController: UIViewController) {
+		guard
+			let opacityDelegate = viewController as? NavigationBarOpacityDelegate,
+			let scrollView = viewController.scrollView
+		else {
+			return
+		}
+		// We can overwrite the existing observer, since Swift 4 block based observers clean themselves up.
+		scrollViewObserver = scrollView.observe(\.contentOffset) { [weak self] _, _ in
+			guard let self = self else { return }
+			guard viewController == self.topViewController else { return }
```

Comment:
Could be simplified:
```suggestion
			guard viewController == self?.topViewController else { return }
```

Reply:
👍  But as this is a 1:1 copy of the AppNavigationController I'll keep this until the refactoring 😉